pkg/alertmanager/distributor.go

```diff
@@ -254,14 +254,36 @@ func (d *Distributor) doUnary(userID string, w http.ResponseWriter, r *http.Requ
 	defer sp.Finish()
 	// Until we have a mechanism to combine the results from multiple alertmanagers,
 	// we forward the request to only only of the alertmanagers.
-	amDesc := replicationSet.Instances[rand.Intn(len(replicationSet.Instances))]
-	resp, err := d.doRequest(ctx, amDesc, req)
-	if err != nil {
-		respondFromError(err, w, logger)
-		return
-	}
+	if req.GetMethod() == "GET" {
```

Comment:
It seems we are just reusing the old code if not GET, which is ok but just duplicate the code

I would prefer to use the same for for both code. I think there are a few options like changing the size of the array to 1 if it is not a GET or have a retry counter which can be different for each api method.

nit: Also i would probably not check for 
`i == len(instances)-1`
i think we can keep a hold of the last error and throw it if the for finish without succeeds.

Reply:
Modified the logic to change the size of the array to 1 if it is not a get request.
for the last comment `keep a hold of the last error and throw it if the for loop finish without succeeds.` 
this check `if err != nil && i == len(instances)-1 {` also does the same thing right. can you let me know if I am missing something here?